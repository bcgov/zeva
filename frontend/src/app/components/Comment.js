import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';
import moment from 'moment-timezone';

const Comment = (props) => {
  const {
    comment, date, user, commentArray,
  } = props;

  return (
    <div className="bcgov-callout" role="alert">
      <span>
        <b>Comments &mdash;</b> {user && date && ({ user }, { date })}
        {comment && (<p className="mt-3">{comment}</p>)}
        {commentArray && (
          commentArray.map((each) => (
            <p key={each.id}>
              {each.createUser.displayName}, {moment(each.createTimestamp).format('YYYY-MM-DD h[:]mm a')}: {each.comment}
            </p>
          ))
        )}
      </span>
    </div>
  );
};

Comment.defaultProps = {
  date: '',
  user: '',
  comment: '',
  commentArray: [],
};
Comment.propTypes = {
  date: PropTypes.string,
  user: PropTypes.string,
  comment: PropTypes.string,
  commentArray: PropTypes.arrayOf(PropTypes.shape()),
};
export default Comment;
