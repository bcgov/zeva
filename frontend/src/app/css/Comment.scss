.bcgov-callout {
    padding: 25px;
    border-left: 10px solid #38598a;
    margin: 16px 0;
    background-color: #f2f2f2;
    }
    
    .bcgov-callout h1, h2, h3, h4 {
    margin: 0;
    }

#comment-input {
    .ql-container {
        height: auto !important;
      }
      .ql-editor {
        min-height: 150px !important;
        max-height: 300px;
        overflow: hidden;
        overflow-y: scroll;
        overflow-x: scroll;
      }
      .comment-box {
        border: 1px solid $border-grey;
        width: 100%;
        padding: 0.5rem 0.5rem;
      }
}