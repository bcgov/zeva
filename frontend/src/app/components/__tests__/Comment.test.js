import React from 'react';
import Comment from '../Comment';
import renderer from 'react-test-renderer'

describe('Comment component', () => {
  it('renders correctly with comments', () => {
    const commentArray = [
      {
        id: 1,
        createUser: { displayName: 'John Doe' },
        createTimestamp: '2022-01-01T01:00:00.000Z',
        comment: 'A sample comment'
      },
      {
        id: 2,
        createUser: { displayName: 'Jane Doe' },
        createTimestamp: '2022-01-02T01:00:00.000Z',
        comment: 'Another sample comment'
      }
    ];
    const component = renderer.create(<Comment commentArray={commentArray} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with empty commentArray', () => {
    const component = renderer.create(<Comment />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
