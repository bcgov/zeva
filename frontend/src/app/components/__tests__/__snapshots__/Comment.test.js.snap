// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component renders correctly with comments 1`] = `
<div
  className="bcgov-callout"
  role="alert"
>
  <span>
    <p>
      John Doe
      ,
       
      2021-12-31 5:00 pm
      :
       
      A sample comment
    </p>
    <p>
      Jane Doe
      ,
       
      2022-01-01 5:00 pm
      :
       
      Another sample comment
    </p>
  </span>
</div>
`;

exports[`Comment component renders correctly with empty commentArray 1`] = `
<div
  className="bcgov-callout"
  role="alert"
>
  <span />
</div>
`;
